codec-http/src/main/java/io/netty/handler/codec/http/HttpHeaderDateFormatter.java

```diff
@@ -0,0 +1,443 @@
+        int len = tokenEnd - tokenStart;
+
+        if (len == 2) {
+            char c0 = txt.charAt(tokenStart);
+            char c1 = txt.charAt(tokenStart + 1);
+            if (isDigit(c0) && isDigit(c1)) {
+                year = getNumericalValue(c0) * 10 + getNumericalValue(c1);
+                return true;
+            }
+
+        } else if (len == 4) {
+            char c0 = txt.charAt(tokenStart);
+            char c1 = txt.charAt(tokenStart + 1);
+            char c2 = txt.charAt(tokenStart + 2);
+            char c3 = txt.charAt(tokenStart + 3);
+            if (isDigit(c0) && isDigit(c1) && isDigit(c2) && isDigit(c3)) {
+                year = getNumericalValue(c0) * 1000 +
+                        getNumericalValue(c1) * 100 +
+                        getNumericalValue(c2) * 10 +
+                        getNumericalValue(c3);
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private boolean parseToken(CharSequence txt, int tokenStart, int tokenEnd) {
+        // return true if all parts are found
+        if (!timeFound) {
+            timeFound = tryParseTime(txt, tokenStart, tokenEnd);
+            if (timeFound) {
+                return dayOfMonthFound && monthFound && yearFound;
+            }
+        }
+
+        if (!dayOfMonthFound) {
+            dayOfMonthFound = tryParseDayOfMonth(txt, tokenStart, tokenEnd);
+            if (dayOfMonthFound) {
+                return timeFound && monthFound && yearFound;
+            }
+        }
+
+        if (!monthFound) {
+            monthFound = tryParseMonth(txt, tokenStart, tokenEnd);
+            if (monthFound) {
+                return timeFound && dayOfMonthFound && yearFound;
+            }
+        }
+
+        if (!yearFound) {
+            yearFound = tryParseYear(txt, tokenStart, tokenEnd);
+        }
+        return timeFound && dayOfMonthFound && monthFound && yearFound;
+    }
+
+    private Date parse0(CharSequence txt, int start, int end) {
+        boolean allPartsFound = parse1(txt, start, end);
+        return allPartsFound && normalizeAndValidate() ? computeDate() : null;
+    }
+
+    private boolean parse1(CharSequence txt, int start, int end) {
+        // return true if all parts are found
+        int tokenStart = -1;
+
+        for (int i = start; i < end; i++) {
+            char c = txt.charAt(i);
+
+            if (isDelim(c)) {
+                if (tokenStart != -1) {
+                    // terminate token
+                    if (parseToken(txt, tokenStart, i)) {
+                        return true;
+                    }
+                    tokenStart = -1;
+                }
+            } else if (tokenStart == -1) {
+                // start new token
+                tokenStart = i;
+            }
+        }
+
+        // terminate trailing token
+        return tokenStart != -1 ? parseToken(txt, tokenStart, txt.length()) : false;
```

Comment:
nah I do this all the time ... helps to be reminded periodically ... thats what reviews are for :)